internal/toolexec/aspect/resolve.go

```diff
@@ -19,79 +19,61 @@ import (
 	args, err := prepareGoCommandArgs("list", "-toolexec", toolexec, "-json", "-deps", "-export", "--", importPath)
 	if err != nil {
 		return nil, fmt.Errorf("preparing go command %v: %w", args, err)
 	}
-	for {
-		cmd := exec.Command("go", args...)
-		var stdout bytes.Buffer
-		cmd.Stdout = &stdout
-		log.Tracef("Attempting to resolve %q using %q\n", importPath, cmd.Args)
-		if err = cmd.Run(); err != nil {
-			return nil, fmt.Errorf("running %q: %w", cmd.Args, err)
-		}
 
-		type listItem struct {
-			ImportPath string // The import path of the package
-			Export     string // The path to its archive, if any
-			BuildID    string // The build ID for the package
-			Standard   bool   // Whether this is from the standard library
-		}
-		var items []listItem
-		dec := json.NewDecoder(&stdout)
-		for {
-			var item listItem
-			if err := dec.Decode(&item); err == io.EOF {
-				break
-			} else if err != nil {
-				return nil, fmt.Errorf("parsing `go list` output: %w", err)
-			}
-			items = append(items, item)
-		}
-
-		output := make(map[string]string, len(items))
-		var err error
-		for _, item := range items {
-			if item.Standard && item.ImportPath == "unsafe" && item.Export == "" {
-				// Special-casing "unsafe", because it's not provided like other modules
-				continue
-			}
-			if item.Export == "" {
-				err = errors.Join(err, fmt.Errorf("%s (%s) has no export file", item.ImportPath, item.BuildID))
-				continue
-			}
-			output[item.ImportPath] = item.Export
-		}
+	cmd := exec.Command("go", args...)
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	log.Tracef("Attempting to resolve %q using %q\n", importPath, cmd.Args)
+	if err = cmd.Run(); err != nil {
+		return nil, fmt.Errorf("running %q: %w", cmd.Args, err)
+	}
 
-		if err == nil {
-			return output, nil
-		} else if attemptedBuild {
-			return nil, fmt.Errorf("after `go build`: %s", err)
+	type listItem struct {
+		ImportPath string // The import path of the package
+		Export     string // The path to its archive, if any
+		BuildID    string // The build ID for the package
+		Standard   bool   // Whether this is from the standard library
+	}
+	var items []listItem
+	dec := json.NewDecoder(&stdout)
+	for {
+		var item listItem
+		if err := dec.Decode(&item); err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("parsing `go list` output: %w", err)
 		}
+		items = append(items, item)
+	}
 
-		// Not found or stale -- let's try to `go build` it so it's present and not stale.
-		// Retrieve original go command flags and pass them along
-		args, err = prepareGoCommandArgs("build", "-toolexec", toolexec, "--", importPath)
-		if err != nil {
-			return nil, fmt.Errorf("retrieving go command flags: %w", err)
+	output := make(map[string]string, len(items))
+	for _, item := range items {
+		if item.Standard && item.ImportPath == "unsafe" && item.Export == "" {
+			// Special-casing "unsafe", because it's not provided like other modules
+			continue
 		}
```

Comment:
Is `unsafe` really the only module to be like that? What about `import "C"`? There is probably also the `builtins` module on top of my head. Can you extract this logic to make it standout from the rest. Like creating a `virtualPackagesNames` const array or something like this?